### PR TITLE
Adapt to storage 2.0 API changes

### PR DESCRIPTION
**Pull request by chrieke**
Updates to the Storage 2.0 API revamp
- Adjusts endpoints for get-assets, asset-info, asset-download-url
- Adds asset-update-metadata
- Adjusts to new asset metadata conventions

Items:
* [x] Ran tests
* [] Ran live-tests
* [x] Implemented (new) unit tests
* [ ] Removed credentials
* [ ] Removed argument pointing to staging
* [ ] Updated [documentation](sdk.up42.com)

For release:
* [ ] Bumped version
* [ ] Added changelog
* [ ] Updated announcement banner

### CODE REVIEW

File: tests/test_asset.py [hunk 1]
```diff
@@ -34,13 +34,11 @@ def test_asset_info_live(asset_live):
...
-
-
-@pytest.mark.live
-def test_asset_source_live(asset_live):
-    assert asset_live.source == "BLOCK"
+def test_asset_update_metadata(asset_mock):
+    updated_info = asset_mock.update_metadata(
+        title="some_other_title", tags=["othertag1", "othertag2"]
+    )
+    assert updated_info["title"] == "some_other_title"
```
Comment (janchrizz) [inline tests/test_asset.py:41]: i think you should also assert that the tags get updated?

File: tests/test_storage.py [hunk 2]
```diff
@@ -124,38 +124,34 @@ def test_get_assets_pagination(auth_mock, requests_mock):
...
+        "size": 50,
+        "number": 0,
+        "empty": False,
     }
 
     # assets pages
     url_storage_assets_paginated = (
-        f"{auth_mock._endpoint()}/workspaces/{auth_mock.workspace_id}/"
-        f"assets?format=paginated&sort=createdAt,asc&size=50"
+        f"{auth_mock._endpoint()}/v2/assets?format=paginated&sort=created,asc&size=50"
```
Comment (janchrizz) [inline tests/test_storage.py:149]: i think `format=paginated` is not necessary

File: up42/asset.py [hunk 3]
```diff
@@ -33,33 +32,37 @@ def __init__(self, auth: Auth, asset_id: str, asset_info: Optional[dict] = None)
...
+        response_json = self.auth._request(
+            request_type="POST", url=url, data=body_update
+        )
+        self._info = response_json
+        return self._info
 
     def _get_download_url(self) -> str:
-        url = f"{self.auth._endpoint()}/workspaces/{self.workspace_id}/assets/{self.asset_id}/downloadUrl"
+        url = f"{self.auth._endpoint()}/v2/assets/{self.asset_id}/download-url"
         response_json = self.auth._request(request_type="GET", url=url)
```
Comment (janchrizz) [inline up42/asset.py:66]: request_type is POST

File: up42/storage.py [hunk 4]
```diff
@@ -63,34 +68,42 @@ def _query_paginated(
...
             Asset objects in the workspace or alternatively json info of the assets.
         """
         allowed_sorting_criteria = [
-            "createdAt",
-            "updatedAt",
-            "source",
+            "created",
+            "updated",
+            "workspaceId",
             "type",
```
Comment (janchrizz) [inline up42/storage.py:107]: i dont think sort by "type" is a thing

**Review by janchrizz — CHANGES_REQUESTED** [review]
Please see individual comments

File: up42/storage.py [hunk 5]
```diff
@@ -63,34 +68,42 @@ def _query_paginated(
...
-            sortby: The sorting criteria, one of "createdAt", "updatedAt", "source", "type",
-                "name", "size".
+            sortby: The sorting criteria, one of "created", "updated", "workspaceId", "name", "size".
             descending: The sorting order, True for descending (default), False for ascending.
+            return_json: If set to True, returns json object.
 
         Returns:
             Asset objects in the workspace or alternatively json info of the assets.
         """
         allowed_sorting_criteria = [
```
Comment (janchrizz) [inline up42/storage.py:103]: also we have sort by producerName and collectionName

File: tests/test_storage.py (hunk 2 above)
Comment (chrieke) [reply tests/test_storage.py:149]: API v2 is paginated by default it seems? Cool!

File: up42/storage.py (hunk 5 above)
Comment (chrieke) [reply up42/storage.py:103]: Added new sorting parameters to docstring, removed code limitation.

**Review by janchrizz — APPROVED** [review]
LGTM

**Pull request merged**